src/votekit/pref_profile/pref_profile.py

```diff
@@ -938,25 +967,25 @@ def __set_class_attrs_from_df(
                 "contains_scores is True but we found no ballots with scores."
             )
 
-        return self
-
-    @model_validator(mode="after")
-    def ballot_list_to_df(self) -> Self:
+    def _ballot_list_to_df(self, ballots: tuple[Ballot, ...]):
```

Comment:
```suggestion
    def _save_ballot_list_to_internal_df(self, ballots: tuple[Ballot, ...]) -> None:
```

Maybe consider doing this without side-effects?